src/xcode/ENA/ENA/Source/Extensions/__tests__/ENTemporaryExposureKey+ProcessingTests.swift

```diff
@@ -23,317 +23,422 @@ import ExposureNotification
 import XCTest
 
 final class ExposureKeysProcessingTests: XCTestCase {
-
-	func testSubmissionPreprocess_ApplyNoInformationVector() {
+	
+	func testSubmissionPreprocess_ApplyNoInformationVectors() {
 		let symptomsOnset: SymptomsOnset = .noInformation
 		let transmissionRiskVector = symptomsOnset.transmissionRiskVector
-
+		let daysSinceOnsetOfSymptomsVector = symptomsOnset.daysSinceOnsetOfSymptomsVector
+		
 		let days = 0...14
-		var keys = days.map { makeMockKey(daysUntilToday: $0) }
-		keys = keys.processedForSubmission(with: symptomsOnset, today: Date(timeIntervalSinceReferenceDate: 0))
-
-		keys.sort { $0.rollingStartNumber > $1.rollingStartNumber }
-
-		for (key, day) in zip(keys, days) {
+		let keys = days.map { makeMockKey(daysUntilToday: $0) }
+		var processedKeys = keys.processedForSubmission(with: symptomsOnset, today: Date(timeIntervalSinceReferenceDate: 0))
+		
+		processedKeys.sort { $0.rollingStartIntervalNumber > $1.rollingStartIntervalNumber }
+		
+		for (key, day) in zip(processedKeys, days) {
```

Comment:
The risk I see with this test approach is that when you change something in SymptomsOnset, the tests will pass. This defies the purpose of having to change both prod and test code to make a new feature pass. 

While checking each value explicitly for all scenario might be too much, I would suggest to add spot checks on TRL and DSOS for maybe the first, last, and some element in the array. The expected value should be hard-coded in the test code.
I would prefer if we had specific spot checks in addition to just looping over this.

Reply:
I agree that a change of values in SymptomsOnset should trigger a test to fail. We have separate tests that cover every single value in SymptomsOnsetTests.swift, that's why I decided not to check them explicitly again in the tests where the values are just applied. Do you agree with me, that the combination of tests are enough to cover double-entry bookkeeping or do you think those tests are not sufficient?

Reply:
Good counter-argument 👍
That makes me think whether it makes sense that all test cases of SymptomsOnsetTests should be moved instead and treat the fact that SymptomsOnset exists as a "design decision" that in itself does not have to be tested.
I'm a big supported of automated unit tests but I don't see it so religiously that every class needs to have its own test class. If a new class exists only because of a certain design decision and it's basically only used in one context (i.e. `processedForSubmission`), I think that's fine.
Anyway, massive duplication of the test logic here and in SymptomsOnsetTests should be avoided for sure. Benefit of having everything here would be that you could see at a glance the values for TRL, for DSOS, etc.
What do you think?

Reply:
I would like to keep them independently. I don't see a huge benefit to bring these two sets of tests together if we can cleanly separate them and already have 😅

Reply:
Fair enough 👍